skia/RSkComponentBVLinearGradient.cpp

```diff
@@ -0,0 +1,131 @@
+  RNS_LOG_NOT_IMPL;
+  return;
+}
+
+RnsShell::LayerInvalidateMask  RSkComponentBVLinearGradient::updateComponentProps(SharedProps newViewProps, bool forceUpdate) {
+  RnsShell::LayerInvalidateMask invalidateMask = RnsShell::LayerInvalidateNone;
+
+  auto const &newBVLinearGradientProps = *std::static_pointer_cast<LinearGradientViewProps const>(newViewProps);
+  auto const &oldBVLinearGradientProps = *std::static_pointer_cast<LinearGradientViewProps const>(getComponentData().props);
+
+  RNS_LOG_INFO(" Start: [" << newBVLinearGradientProps.start.x << "," << newBVLinearGradientProps.start.y <<
+                  "], End: [" << newBVLinearGradientProps.end.x << "," << newBVLinearGradientProps.end.y <<
+                  "], Colors: " << newBVLinearGradientProps.colors.size() <<
+                  ", Locations: " << newBVLinearGradientProps.locations.size() <<
+                  ", UseAngle: " << newBVLinearGradientProps.useAngle <<
+                  ", Angle: " << newBVLinearGradientProps.angle <<
+                  ", AngleCenter: [" << newBVLinearGradientProps.angleCenter.x << "," << newBVLinearGradientProps.angleCenter.y << "]");
+
+  RNS_UNUSED(newBVLinearGradientProps);
+  RNS_UNUSED(oldBVLinearGradientProps);
+  return invalidateMask;
+}
+
+void RSkComponentBVLinearGradient::OnPaint(SkCanvas *canvas) {
+
+  auto component = getComponentData();
+  auto const &linearGradientProps = *std::static_pointer_cast<LinearGradientViewProps const>(component.props);
+
+  /* apply view style props */
+  auto borderMetrics=linearGradientProps.resolveBorderMetrics(component.layoutMetrics);
+  size_t numColors = linearGradientProps.colors.size();
+
+  Rect frame = component.layoutMetrics.frame;
+  SharedColor bgColour = linearGradientProps.backgroundColor;
+  SkScalar frameX = frame.origin.x;
+  SkScalar frameY = frame.origin.y;
+  SkScalar frameWidth = frame.size.width;
+  SkScalar frameHeight = frame.size.height;
+
+  // If background color is transparent then need to find atleast one non-transparent colour in gradient colours so that we can draw shadow properly.
+  for(int i = 0; (!(bgColour && (colorComponentsFromColor(bgColour).alpha != 0.0)) && i < numColors); i++) {
+    bgColour = linearGradientProps.colors.at(i);
+  }
+
+  /*Draw Order : 1. Shadow 2. Gradient 3 Border*/
+  auto layerRef=layer();
+  if(layerRef->isShadowVisible) {
+    drawShadow(canvas,frame,borderMetrics,
+                bgColour, // Used only to decide on what type of shadow to draw.
+                layerRef->shadowColor,layerRef->shadowOffset,layerRef->shadowOpacity,
+                layerRef->opacity,
+                layerRef->shadowImageFilter,layerRef->shadowMaskFilter);
+  }
+
+  RNS_LOG_DEBUG(std::this_thread::get_id() << "] Layer " << layer().get() <<" Draw Picture Here with Frame  [" << frameX << "," << frameY << "," << frameWidth << "," << frameHeight << "]");
+
+  SkColor colors[numColors];
+  SkScalar pos[linearGradientProps.locations.size()];
```

Comment:
Is it possible App allows to specify only Single against the expectation, If then can the shader creation can be avoided under colorSize > 1 condition ?